### ISSUE TITLE
False Positive: IllegalInstantiationCheck does not recognize records shadowing java.lang.Boolean

### ISSUE THREAD

**Issue by lithops-zty** [checkstyle/checkstyle#21331]
I have read check documentation: https://checkstyle.org/checks/coding/illegalinstantiation.html
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
/var/tmp $ javac Boolean.java

/var/tmp $ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
    "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
    "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="IllegalInstantiation">
      <property name="classes" value="java.lang.Boolean"/>
    </module>
  </module>
</module>

/var/tmp $ cat Boolean.java
record Boolean(int value) {
    static Boolean of(int v) { return new Boolean(v); }
}

/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-13.5.0-all.jar -c config.xml Boolean.java
Starting audit...
[ERROR] /var/tmp/Boolean.java:2:39: Instantiation of java.lang.Boolean should be avoided. [IllegalInstantiation]
Audit done.
Checkstyle ends with 1 errors.
```

---

**Describe what you expect in detail.**

No violation should be reported. `new Boolean(v)` refers to the record `Boolean` declared in the same file, which shadows `java.lang.Boolean`. The check only considers custom shadowing `CLASS_DEF` names but not `RECORD_DEF`.

**Comment by lithops-zty** [comment]
On it.

**Comment by romani** [comment]
Fix is merged
